src/lib.rs

```diff
@@ -206,6 +221,28 @@ impl AstarteSdk {
         Ok(())
     }
 
+    pub async fn add_interface(&self, file_path: &str) -> Result<(), AstarteError> {
+        let path = Path::new(file_path);
+        match Interface::from_file(path) {
+            Ok(interface) => {
+                let interfaces_mutex = self.interfaces.clone();
+                let mut interfaces = interfaces_mutex.write().await;
+                let interfaces_map = &mut interfaces.interfaces;
+                if interface.get_ownership() == interface::Ownership::Server {
+                    self.subscribe_server_owned_interface(&interface).await?;
+                }
```

Comment:
This can be moved to _before_ we acquire the lock since it doesn't need to be in the critical section